### PR TITLE
Bumping ruby version

### PR DESCRIPTION
**Pull request by kyledurand**
@kirs not sure about this travis file but shouldn't be an issue to bump the rvm version as well right?

### CODE REVIEW

File: dev.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ up:
     - openssl
     - sqlite
   - ruby:
-      version: 2.2.3p172-shopify
+      version: 2.3.1
```
Comment (rafaelfranca) [inline dev.yml]: Should not it be 2.3.3? Or greater? We use that in production, so it makes sense to use that here too.

File: .travis.yml [hunk 2]
```diff
@@ -14,11 +14,11 @@ env:
     - SUITE=pg
     - SUITE=sqlite3
 rvm:
-  - 2.2.3
```
Comment (rafaelfranca) [inline .travis.yml]: I'd keep both and also add 2.4 to the list, we want to make sure all ruby versions works unless we have a strong reason to remove support to 2.2.

File: .travis.yml [hunk 3]
```diff
@@ -14,11 +14,13 @@ env:
...
+  - 2.2.6
+  - 2.3.3
+  - 2.4.1
```
Comment (rafaelfranca) [inline .travis.yml]: Urgh, the app don't work with 2.4 yet, so maybe we can leave only 2.2 and 2.3 now?

File: dev.yml [hunk 4]
```diff
@@ -7,7 +7,7 @@ up:
     - openssl
     - sqlite
   - ruby:
-      version: 2.2.3p172-shopify
+      version: 2.3.3
```
Comment (casperisfine) [inline dev.yml]: Please keep the dev version to a 2.x release.

Comment (kirs) [inline dev.yml]: > Shipit should stay 2.2.x compatible for the time being.

Why can't we use 2.3 in dev and keep 2.2 compatibility in Travis?

Comment (casperisfine) [inline dev.yml]: - People would use 2.3 feature in dev just to be surprised by the CI breaking when they submit their PR
- What advantage would that give us?

Comment (kyledurand) [inline dev.yml]: I couldn't dev up using `2.2.3p172-shopify` which is why I proposed this change. If it's going to cause issues I can work around that and close this PR

Comment (kirs) [inline dev.yml]: I think what was meant was pure 2.2.3 without shopify additions

**Pull request merged**